addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.satel.internal.discovery;
+
+import static org.openhab.binding.satel.SatelBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.satel.handler.SatelBridgeHandler;
+import org.openhab.binding.satel.internal.BundleTextTranslator;
+import org.openhab.binding.satel.internal.command.ReadDeviceInfoCommand;
+import org.openhab.binding.satel.internal.command.ReadDeviceInfoCommand.DeviceType;
+import org.openhab.binding.satel.internal.command.SatelCommand;
+import org.openhab.binding.satel.internal.config.SatelThingConfig;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link SatelDeviceDiscoveryService} searches for available Satel
+ * devices and modules connected to the API console
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ *
+ */
+public class SatelDeviceDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int OUTPUT_FUNCTION_SHUTTER = 105;
+
+    private final Logger logger = LoggerFactory.getLogger(SatelDeviceDiscoveryService.class);
+
+    private SatelBridgeHandler bridgeHandler;
+    private BundleContext bundleContext;
+    private volatile boolean scanStopped;
+
+    public SatelDeviceDiscoveryService(SatelBridgeHandler bridgeHandler, BundleContext bundleContext)
+            throws IllegalArgumentException {
```

Comment:
This a runtime exception no need and no valuable information by adding this throws it can also throw an `ArrayIndexOutOfBoundsException` and an `NullPointerException`

Reply:
So why is it declared in the super class? :) 
Anyway you're right, removed.